### PR TITLE
CustomPages: Add core isSuccess Method(s)

### PR DESCRIPTION
**Pull request by kingthorin**
- Add `isSuccess(HttpMessage)` in `AbstractPlugin` and `PassiveScanData`.
- Add Unit Tests for the new method.
- Tweak javadoc for `isClientError(HttpMessage)` and `isServerError(HttpMessage)` in both classes as well.

Related to zaproxy/zaproxy#9

### CODE REVIEW

File: zap/src/main/java/org/zaproxy/zap/extension/pscan/PassiveScanData.java [hunk 1]
```diff
@@ -247,8 +247,31 @@ public boolean isPageOther(HttpMessage msg) {
...
+        if (isCustomPage(msg, CustomPage.Type.NOTFOUND_404)
+                || isCustomPage(msg, CustomPage.Type.ERROR_500)) {
+            return false;
+        }
+        if (isCustomPage(msg, CustomPage.Type.OK_200)) {
+            return true;
+        }
+        return HttpStatusCode.isSuccess(
+                msg.getResponseHeader()
+                        .getStatusCode()); // TODO JavaDoc and dbl check then UnitTests.
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/extension/pscan/PassiveScanData.java:269]: Remove?

Comment (kingthorin) [reply zap/src/main/java/org/zaproxy/zap/extension/pscan/PassiveScanData.java:269]: Thanks, was a placeholder when I had to walk away at one point.

**Comment by kingthorin** [comment]
Spurious TODO comment removed.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thanks!

**Review by psiinon — APPROVED** [review]

**Pull request merged**